File: app/src/main/java/com/example/a11699/androidxallstudy/piaodanmu/MyTextView.java
package com.example.a11699.androidxallstudy.piaodanmu;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.TextView;

import androidx.appcompat.widget.AppCompatTextView;

/**
 * @Author Yu
 * @Date 2021/6/24 20:13
 * @Description TODO
 */
public class MyTextView extends AppCompatTextView {
    public MyTextView(Context context) {
        super(context);
    }
    public MyTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }
    public MyTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }
    //返回textview是否处在选中的状态
    //而只有选中的textview才能够实现跑马灯效果
    @Override
    public boolean isFocused() {
        return true;
    }
}

File: comp_customview/src/main/java/com/example/a11699/comp_customview/MyTextView.kt
package com.example.a11699.comp_customview

import android.content.Context
import android.graphics.Canvas
import android.util.AttributeSet
import android.view.View

/**
 *Create time 2020/6/26
 *Create Yu
 *description:
 */
class MyTextView : View {
    constructor(context: Context) : super(context) {
    }
    constructor(context: Context,attributeSet: AttributeSet) : super(context,attributeSet) {
    }
    constructor(context: Context,attributeSet: AttributeSet,detStyleAttr:Int) : super(context,attributeSet,detStyleAttr) {
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        var widthMode = MeasureSpec.getMode(widthMeasureSpec)
        var heightMode = MeasureSpec.getMode(heightMeasureSpec)

        when(widthMode){
            MeasureSpec.AT_MOST->{
                //对应 wrap_content
            }
            MeasureSpec.EXACTLY->{
                //指定确切值 100dp match_parent fill_parent
            }
            MeasureSpec.UNSPECIFIED->{
                //尽可能的大 很少用到 ListView ScrollView会用到
           }
        }
    }

    override fun onDraw(canvas: Canvas?) {
        super.onDraw(canvas)

    }
}